src/bentoml/_internal/frameworks/onnx.py

```diff
@@ -289,21 +296,24 @@ def forward(self, x, bias):
                 f"Provided method names {[m for m  in provided_methods if m != 'run']} are invalid. 'bentoml.onnx' will load ONNX model into an 'onnxruntime.InferenceSession' for inference, so the only supported method name is 'run'."
             )
 
-    if not isinstance(model, onnx.ModelProto):
-        raise TypeError(f"Given model ({model}) is not a onnx.ModelProto.")
+    run_sig: ModelSignature | ModelSignatureDict = signatures["run"] or {}
+    run_sig_dict: ModelSignatureDict = (
+        run_sig
+        if isinstance(run_sig, dict)
+        else t.cast(ModelSignatureDict, attr.asdict(run_sig))
+    )
 
-    if len(model.graph.output) > 1:
-        logger.warning(
-            "The model you are attempting to save has more than one output. The ONNX runner will only return the first output."
-        )
+    run_sig_dict["input_spec"] = [MessageToDict(inp) for inp in model.graph.input]
+    run_sig_dict["output_spec"] = [MessageToDict(out) for out in model.graph.output]
```

Comment:
Here I'm not sure if we want to use `input_spec`/`output_spec` field now or we want to reserve these fields until we have unified io spec for runner interface and use some temporary fields to save onnx spec dict for now.

Reply:
but these specs really belongs to `input_spec/output_spec`

Reply:
move to ONNXOptions for now